docs/developers.rst

```diff
@@ -43,16 +43,20 @@ Publish a release
 
 To publish a new release on `PyPI`_, make sure the changelog is up to date
 and make sure you bumped the module version in ``setup.py``. Tag master
-at the version. For example, something like::
+as that version. For example, something like::
 
     git tag 0.0.5
     git push --tags
 
-Run this from the repository root to publish on `PyPI`_::
+Make sure you have `twine`_ installed, and then run this from the repository
+root to publish on `PyPI`_ as both a source distribution and wheel::
 
-    python setup.py sdist register upload
+    rm -rf dist/*
```

Comment:
Clean only deletes the build directory, not the contents of `dist/` unfortunately.

```python
[~/src/hawkrest]$ python setup.py sdist bdist_wheel
running sdist
...
creating build/bdist.mingw32_nt-6.3-2.4.0(0.292/5/3)-x86_64/wheel/hawkrest-0.0.9.dist-info/WHEEL
[~/src/hawkrest]$ sed -i -e "s/version='0\.0\.9'/version='0\.0\.10'/" setup.py
[~/src/hawkrest]$ python setup.py clean sdist bdist_wheel
running clean
running sdist
...
creating build/bdist.mingw32_nt-6.3-2.4.0(0.292/5/3)-x86_64/wheel/hawkrest-0.0.10.dist-info/WHEEL
[~/src/hawkrest]$ ll dist/
total 32K
-rw-r--r--+ 1 Ed None 4.8K Jan 12 09:35 hawkrest-0.0.10.tar.gz
-rw-r--r--+ 1 Ed None 6.7K Jan 12 09:35 hawkrest-0.0.10-py2.py3-none-any.whl
-rw-r--r--+ 1 Ed None 4.7K Jan 12 09:33 hawkrest-0.0.9.tar.gz
-rw-r--r--+ 1 Ed None 6.7K Jan 12 09:33 hawkrest-0.0.9-py2.py3-none-any.whl
```

This would then mean twine would try to upload both old and new versions, causing an error given v0.0.9 would already exist on PyPI.

Even `clean --all` doesn't help:
```python
[~/src/hawkrest]$ python setup.py clean --all
running clean
removing 'build/lib' (and everything under it)
removing 'build/bdist.mingw32_nt-6.3-2.4.0(0.292/5/3)-x86_64' (and everything under it)
'build/scripts-2.7' does not exist -- can't clean it
[~/src/hawkrest]$ ll dist/
total 32K
-rw-r--r--+ 1 Ed None 4.8K Jan 12 09:35 hawkrest-0.0.10.tar.gz
-rw-r--r--+ 1 Ed None 6.7K Jan 12 09:35 hawkrest-0.0.10-py2.py3-none-any.whl
-rw-r--r--+ 1 Ed None 4.7K Jan 12 09:33 hawkrest-0.0.9.tar.gz
-rw-r--r--+ 1 Ed None 6.7K Jan 12 09:33 hawkrest-0.0.9-py2.py3-none-any.whl
```